data_model/src/lib.rs

```diff
@@ -696,10 +744,21 @@ where
 }
 
 /// This trait marks entity that implement it as identifiable with an `Id` type to find them by.
-pub trait Identifiable: Debug + Clone {
+pub trait Identifiable: Debug {
     /// Type of entity's identification.
     type Id: Into<IdBox> + Debug + Clone + Eq + Ord;
-    /// Type used to register `Identifiable` entity
+
+    fn id(&self) -> &Self::Id;
+}
+
+pub trait HasMetadata {
+    // type Metadata = metadata::Metadata;
+    // Uncomment when stable.
```

Comment:
```suggestion
    // TODO: Uncomment when stable.
```

you're saying we could have more fine grained `Metadata` types? Because not all types having metadata allow for the same `Value` types. If that's the case I'm all for it

Reply:
GAT FTW.

Reply:
We want to turn Metadata into a trait, that's implemented for certain value enums. The overall `Value` enum can be cast into the specific kind with type checking. This adds strong typing to kinds of metadata. 

Moreover, this will pave the way to more precise metadata e.g. exchange rates, accounts and such.

Reply:
exactly, I agree. Though I think we can do this without GAT support